### PR TITLE
[cef-build] Create a new port

### LINKED ISSUE #332: Add CEF
Anybody interested by a [CEF](https://bitbucket.org/chromiumembedded/cef) port?
It's a framework for embedding Chromium web views in native apps.

### PR DESCRIPTION
**Pull request by serg06**
**Describe the pull request**

[CEF](https://bitbucket.org/chromiumembedded/cef/src/master/) was requested 5 years ago but nobody has added it yet: #332 

This is the next best thing: [Pre-built CEF binaries](https://cef-builds.spotifycdn.com/index.html).

#### What does your PR fix?  

Fixes #332 (for win64 builds.)

#### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

x64-windows, Yes

(There are binaries for other triplets on their website, but I didn't implement that.)

#### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes

#### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by serg06** [comment]
Update: I realize I'm not copying the `dll`, `bin`, `pak` and `dat` files correctly. It only ran for me because they were left over from an old build.

How do I get the `pak` and `dat` files to end up in the .exe directory?

File: ports/cef-build/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+file(
+	COPY "${SOURCE_PATH}/include"
+	DESTINATION "${CURRENT_PACKAGES_DIR}"
+)
+
+# Another /include for cef's own imports
+file(
+	COPY "${SOURCE_PATH}/include"
+	DESTINATION "${CURRENT_PACKAGES_DIR}/include"
+)
```
Comment (JackBoosY) [inline ports/cef-build/portfile.cmake:89]: Why not use `vcpkg_install_cmake` instead? Does the source have `install` code?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/cef-build/portfile.cmake (hunk 1 above)
Comment (serg06) [reply ports/cef-build/portfile.cmake:89]: @JackBoosY It doesn't have `install` code unfortunately, I tried `vcpkg_install_cmake` but it told me "invalid target install".

Comment (JackBoosY) [reply ports/cef-build/portfile.cmake:89]: Yeah, therefore, we need to patch the source to add the install command.

**Comment by Pospelove** [comment]
Please take a look at https://github.com/microsoft/vcpkg/pull/10102

**Comment by serg06** [comment]
@Pospelove But doesn't the recent [chartdir port](https://github.com/microsoft/vcpkg/blob/master/ports/chartdir/portfile.cmake) use prebuilts?

**Comment by serg06** [comment]
@JackBoosY I changed it to use `INSTALL()` command for everything, but I still have the same issue, I don't know how to copy `.bin` files to output directory.

For example I added this to CMakeLists.txt:

```
file(GLOB bins_deb Debug/*.dll Debug/*.bin)
file(GLOB bins_rel Release/*.dll Release/*.bin)
INSTALL(
  FILES ${bins_deb}
  CONFIGURATIONS Debug
  TYPE BIN
)
```

And they appear in the bin folder:

image

But they don't get copied over to .exe directory. Am I doing something wrong?

**Comment by JackBoosY** [comment]
@serg06  I will give you some help.

**Comment by JonLiu1993** [comment]
@serg06 ,Can you please resolve the conflicts against master?

**Comment by JackBoosY** [comment]
I noticed that this port only generates the library `libcef_dll_wrapper`, and other libraries copy existing files. Right?
If it's true, we hope that the main library `cef` is generated by the build process and change the port name to cef.

**Comment by serg06** [comment]
@JackBoosY Yeah, this library copies a bunch of pre-built binaries and generates `libcef_dll_wrapper`.

Are you suggesting we generate the `cef` binaries ourselves instead of using prebuilts? How could that help with the current issue?

**Comment by JackBoosY** [comment]
@serg06 According to our policy, we need to generate binaries as much as possible instead of directly installing pre-built binaries.

**Comment by serg06** [comment]
@JackBoosY That's fine, but it has been years and nobody has added cef yet. Wouldn't it better to have prebuilt binaries than to have nothing at all?

**Comment by JackBoosY** [comment]
@serg06 Sorry but according to our policy, only SDK can use pre-built binaries.
We should use vcpkg_configure_gn to build it.
Example: [chromium-base](https://github.com/microsoft/vcpkg/blob/master/ports/chromium-base/portfile.cmake)

**Comment by serg06** [comment]
@JackBoosY Ah okay. Thanks for the help. I hope someone will add it in the future.

**Pull request closed without merging**